.github/workflows/protobuf-reprolang.yml

```diff
@@ -0,0 +1,37 @@
+name: Generated code is up to date
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/**'
+      - 'docs/**'
+      - 'bindings/**'
+      - 'scip.proto'
+      - 'buf**'
+      - '.tool-versions'
+      - 'dev/proto-generate.sh'
+      - 'dev/proto-generate-in-docker.sh'
+      - 'Dockerfile.bindings'
+      - 'cmd/scip/tests/reprolang/**'
+
+jobs:
+  gen-up-to-date:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - run: docker pull ghcr.io/sourcegraph/scip-bindings-env:latest || echo "no suitable cache"
+
+      - name: Regenerate protobuf bindings and reprolang parser
+        run: |
+          # We're changing the owner of the checkout folder to a particular user id,
+          # matching the user id of `asdf` user we create inside the docker container.
+          sudo chown -R 1001:1001 . && ./dev/generate-all-in-docker.sh
```

Comment:
Is this needed because otherwise we might get permission errors inside the container? I don't understand why we need to create a separate `asdf` user inside the container.

Reply:
The reason for this is that I failed to make asdf work with a root user - there were some errors coming out from asdf itself which I didn't have time to debug.

It's an eyesore to have to do this, but to avoid it we'd need to make asdf work with root – which might not be worth the time, given the entire docker container is just a minimal hack to get back to reproducibility.